### PR TITLE
Ss 180

### PR DESCRIPTION
**Pull request by michael-lts**
**Handle multiple sets**
* * *

**JIRA Ticket**: [(link)](https://jira.huit.harvard.edu/browse/SS-182)

* Additional jira: https://jira.huit.harvard.edu/browse/SS-181

# What does this Pull Request do?
iterates through multiple sets, using configuration file for set information

# How should this be tested?
Deploy to dev (done)
Locally, from project root, run > python3 test/invoke_task.py
Look for transformed files in /docker/jstorforum/PROCESSING/harvested
# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests?
- integration tests? Partial, subsequent pr will handle testing of harvest

### CODE REVIEW

**Review by cgoines — APPROVED** [review]

**Comment by cgoines** [comment]
int tests pass

File: app/jstor_harvester.py [hunk 1]
```diff
@@ -42,28 +38,64 @@ def do_task(self, request_json):
...
+                        opDir = set["opDir"]
+                        if not os.path.exists(harvestDir + opDir):
+                            os.makedirs(harvestDir + opDir)
+                        current_app.logger.info("Harvesting set:" + setSpec + ", output dir: " + opDir)
+                        sickle = Sickle(os.getenv("jstor_oai_url"))
+                        
+                        records = sickle.ListRecords(metadataPrefix='oai_ssio', set=setSpec)
+                        for item in records:
+                            current_app.logger.info(item.header.identifier)
+                            f = open(harvestDir + opDir + "/" + item.header.identifier + ".xml", "w")
```
Comment (awoods) [inline app/jstor_harvester.py:95]: @michael-lts : here is where you should use `with`, i.e.
```
with open(harvestDir + opDir + "/" + item.header.identifier + ".xml", "w") as f:
...
```

File: app/jstor_harvester.py [hunk 2]
```diff
@@ -42,28 +38,64 @@ def do_task(self, request_json):
...
         sleep(sleep_s)
 
+        jstorforum = False
+        if 'jstorforum' in request_json:
+            current_app.logger.info("running jstorforum harvest")
+            jstorforum = request_json['jstorforum']
+        if jstorforum:
+            with open('harvestjobs.json') as f:
+                harvjobsjson = f.read()
+            # leave during development, actually pulling from config file
```
Comment (awoods) [inline app/jstor_harvester.py:56]: Please create a JIRA ticket for this and add a link to the ticket as a comment here.

File: app/jstor_harvester.py [hunk 3]
```diff
@@ -42,28 +38,64 @@ def do_task(self, request_json):
...
+                                        "opDir": "loebmusic"
+                                    }
+                                ],
+                                "opDir": "SSIO"
+                            }
+                        },
+                        {"jobName": "aspace"}
+                    ]'''
+            harvestconfig = json.loads(harvjobsjson)
+            #current_app.logger.debug("harvestconfig")
```
Comment (awoods) [inline app/jstor_harvester.py:79]: Can we remove this dead code?

**Review by awoods — CHANGES_REQUESTED** [review]

File: app/jstor_harvester.py (hunk 1 above)
Comment (michael-lts) [reply app/jstor_harvester.py:95]: I was planning on a full review of file openings for the components in the next iteration ...

**Review by awoods — APPROVED** [review]

**Pull request merged**